tracing-core/src/callsite.rs

```diff
@@ -169,3 +187,159 @@ impl Hash for Identifier {
         (self.0 as *const dyn Callsite).hash(state)
     }
 }
+
+// ===== impl Registration =====
+
+impl<T> Registration<T> {
+    /// Construct a new `Registration` from some `&'static dyn Callsite`
+    pub const fn new(callsite: T) -> Self {
+        Self {
+            callsite,
+            next: AtomicPtr::new(ptr::null_mut()),
+        }
+    }
+}
+
+impl fmt::Debug for Registration {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Registration").finish()
```

Comment:
nit: might be nice to print the addresses of the fields here:
```suggestion
        f.debug_struct("Registration")
            .field("callsite", &format_args!("{:p}", self.callsite))
            .field("next", &format_args!("{:p}", self.next.load(Ordering::Acquire))
            .finish()
```